### PR TITLE
(not ready for merging) Make task schema support arbitrary links

### PR DESCRIPTION
**Pull request by wlach**
This will be useful for https://bugzilla.mozilla.org/show_bug.cgi?id=1204490

### CODE REVIEW

**Comment by wlach** [comment]
@jonasfj @djmitche Here's a first stab at updating the schema. Do you have any suggestions on how to test my changes?

File: schemas/task.yml [hunk 1]
```diff
@@ -176,6 +176,31 @@ properties:
...
+        type:           array
+        items:
+          type:       object
+          title:      "Link pairs"
+          description: List of title/value link pairs
+          properties:
+            name:
+              description: Human readable description of where link goes
+              type:        string
+              maxLength:   255
```
Comment (djmitche) [inline schemas/task.yml]: why does this need a maximum length?

File: schemas/task.yml [hunk 2]
```diff
@@ -176,6 +176,31 @@ properties:
...
+            url:
+              description: URL of link
+              type:        string
+              format:      uri
+              maxLength:   4096
```
Comment (djmitche) [inline schemas/task.yml]: same..

File: schemas/task.yml (hunk 1 above)
Comment (wlach) [inline schemas/task.yml]: Just following the convention elsewhere in this file

Comment (jonasfj) [inline schemas/task.yml]: maximum lengths are good... they prevent people from doing bad things... Notably, some crazy person like me might one day try to include a data url... which no longer metadata and definitely would be an abuse :)

**Comment by jonasfj** [comment]
For the record, this needs a few tweaks in the code (I suspect), but otherwise the schema seems reasonable.

**Comment by wlach** [comment]
Added some links to the create task test.

**Comment by jonasfj** [comment]
One note:
The current `task.description` property is already declared to be markdown.
Hence, if these links are strictly for human consumption it might be just as fine to include a list of bulletpoints with links in the `task.description` property.

I'm just thinking that `task.description` would allow for nested lists, categories, header lines, etc. It's intended to be a human readable markdown text. Just a thought. At the end of the day, it might nicer to just have the links in the description, where other formatting options is available too.

**Comment by wlach** [comment]
Yeah, I think we're going to go with the description approach for linking back to Treeherder, so closing this PR for now. We can always resurrect it if we decide we want a forest of links for tasks for some other reason. :)

**Pull request closed without merging**